### PR TITLE
Update 02_Natural-Language-Processing-Basics.md

### PR DESCRIPTION
**Pull request by carolina-museum**
Added explanations of one-hot encoding and string distances.

### CODE REVIEW

File: nlp_basics_articles/02_Natural-Language-Processing-Basics.md [hunk 1]
```diff
@@ -73,6 +73,72 @@ Figure 6. Q&A
 ## Basic concepts in NLP
 The basic concepts used in NLP are tokens, positional encoding, embedding, and encoder/decoder.
 
-# tokens
-One token of a sequence is generated from the input sequence, representing the smallest language unit.
+### One-hot encoding
```
Comment (huyfififi) [inline nlp_basics_articles/02_Natural-Language-Processing-Basics.md:76]: This reminds me of when I worked on the Kaggle Titanic. The cabin numbers don't have any inherent numerical meaning, so treating them as numbers isn't the right approach. One way to take the information into account is to create this kind of flag to indicate cabin positions. This usually increases the dimension of the feature vectors, leading to bad accuracy tho

**Review by huyfififi — APPROVED** [review]

File: nlp_basics_articles/02_Natural-Language-Processing-Basics.md (hunk 1 above)
Comment (carolina-museum) [reply nlp_basics_articles/02_Natural-Language-Processing-Basics.md:76]: Oh, now that you mentioned it, I can imagine the accuracy would drop. 
Did you have any other way to represent the cabin numbers? Maybe some compressed version?

**Comment by carolina-museum** [comment]
@huyfififi Thank you for reviewing!

File: nlp_basics_articles/02_Natural-Language-Processing-Basics.md (hunk 1 above)
Comment (huyfififi) [reply nlp_basics_articles/02_Natural-Language-Processing-Basics.md:76]: @carolina-museum I tried grouping the cabins into larger categories such as A, B, and C, but they still didn't improve the overall accuracy. It's been more than 5 years, so I don't remember it precisely tho

**Pull request merged**